pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -430,6 +438,165 @@ func consumerRefMatches(consumer *corev1.ObjectReference, machine *machinev1.Mac
+			Name:      machine.Name + "-user-data",
+			Namespace: machine.Namespace,
+		}
+		return nil
+	}
+
+	// userDataSecret is the pointer ignition originally
+	// created by openshift installer. Fetch it to extract
+	// its content (an URL containg full ignition and a
+	// CA cert for the URL's https connection)
+	userDataSecret := corev1.Secret{}
+	key = client.ObjectKey{
+		Name:      config.UserData.Name,
+		Namespace: config.UserData.Namespace,
+	}
+	err = a.client.Get(ctx, key, &userDataSecret)
+	if err != nil {
+		log.Printf("Cannot get %s secret: %s", config.UserData.Name, err.Error())
+		return err
+	}
+	encodedUserDataBytes := userDataSecret.Data["userData"]
+	var userData []byte
+	userData, err = base64.StdEncoding.DecodeString(string(encodedUserDataBytes))
+	if err != nil {
+		log.Printf("Error base64 decoding userData : %s", err.Error())
+		return fmt.Errorf("error base64 decoding userData: %w", err)
+	}
+
+	// Define an ignition data struct to unmarshal incoming json
+	type IgnitionData struct {
+		Ignition struct {
+			Config struct {
+				Append []struct {
+					Source string
+				}
+			}
+			Security struct {
+				TLS struct {
+					CertificateAuthorities []struct {
+						Source string
+					}
+				}
+			}
+		}
+	}
+	var ignData IgnitionData
+
+	if err := json.Unmarshal(userData, &ignData); err != nil {
+		log.Printf(string(userData))
+		log.Printf("Cannot unmarshal ignition config from userData : %s", err.Error())
+		return fmt.Errorf("could not unmarshal ignition config from userData: %w", err)
+	}
+	// Fetch IgnitionURL and the CA cert
+	ignitionURL := ignData.Ignition.Config.Append[0].Source
+	caCertRaw := ignData.Ignition.Security.TLS.CertificateAuthorities[0].Source
+	caCertB64 := strings.TrimPrefix(caCertRaw, "data:text/plain;charset=utf-8;base64,")
+
+	// We didn't get any URL
+	if ignitionURL == "" {
+		err := fmt.Errorf("error extracting full Ignition URL")
+		log.Printf("URL not found : %s", err.Error())
+		return err
+	}
+	// We didn't get the URL's Cert
+	if caCertB64 == "" {
+		err := fmt.Errorf("CA cert for full Ignition URL was not specified")
+		log.Printf("Cert not found in ignition data: %s", err.Error())
+		return err
+	}
+	caCert, err := base64.StdEncoding.DecodeString(caCertB64)
+	if err != nil {
+		log.Printf("Could not base64 decode caCert: %s", err.Error())
+		return fmt.Errorf("could not base64 decode caCert: %w", err)
+	}
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+	client := retryablehttp.NewClient()
+	client.HTTPClient.Transport = &http.Transport{
+		TLSClientConfig: &tls.Config{
+			RootCAs: caCertPool,
+		},
+	}
+
+	// Get the full ignition from the URL
+	resp, err := client.Get(ignitionURL)
+	if err != nil {
+		log.Printf("Error fetching full ignition from URL : %s", err.Error())
+		return fmt.Errorf("error fetching full ignition from URL: %w", err)
+	}
+	defer resp.Body.Close()
+
+	var fullIgnition []byte
+	fullIgnition, err = ioutil.ReadAll(resp.Body)
```

Comment:
Should check the response code first to make sure it's not a 404, 5xx, etc.

Reply:
Just checking for STATUSOK for now. Anything else will cause error.